src/iterable_recursive_open_struct.rb

```diff
@@ -0,0 +1,26 @@
+
+module Metalware
+  class IterableRecursiveOpenStruct < RecursiveOpenStruct
+    def each(&block)
+      convert_hash_values_to_own_class.each(&block)
+    end
+
+    def each=(*args)
+      raise IterableRecursiveOpenStructPropertyError,
+        "Cannot set property 'each', reserved to use for iteration"
+    end
+
+    private
+
+    def convert_hash_values_to_own_class
```

Comment:
Actually it probably wouldn't matter. If a method was to call `convert_hash_values_to_own_class` it would find the private method. However from outside, it would got to `method_missing`.

So LGTM then